### PR TITLE
DWGUGens: fix float/double confusion for GCC 6 compatibility

### LINKED ISSUE #106: source/DWGUGens/dwglib/DWG.hpp fail to build with GCC 6
The source fail to build in Debian after Debian switched to GCC 6 as the default C++ compiler.  See https://bugs.debian.org/831078 for details.

The problem is that this expression no longer work in source/DWGUGens/dwglib/DWG.hpp:
```
float loss;
...
void set_area(float a1,float lossF)
    {
        if(a1 < 1e-18)
            loss = 0.0;
        else
            loss = 1.0 - std::min(lossF/sqrt(static_cast<double>(a1)),1.0);

    }
```

I believe the cause is that the compiler now pick the function float sqrt(float) instead of the function double sqrt(double).  There are two ways to solve the build problem, and I am unsure which one is the best.  Either replace 1.0 with 1.0f to use float everywhere in the calculations, or cast a1 using static_cast(double>(a1) to tell the compiler to use doubles everywhere.  I assume the former calculation is slightly quicker, while the latter is more accurate.

Do you have an opinion on how this should be solved?  How important is the accuracy of the calculation?

### PR DESCRIPTION
**Pull request by snappizz**
Fix #106 

I can't reproduce the original bug but I can confirm that this doesn't cause any problems

### CODE REVIEW

**Comment by petterreinholdtsen** [comment]
Note, this is the error from gcc 6:

```
In file included from /home/pere/src/sonic-pi/pkg-supercollider-sc3-plugins/source/DWGUGens/dwglib/DWG.cpp:19:0,
                 from /home/pere/src/sonic-pi/pkg-supercollider-sc3-plugins/source/DWGUGens/DWGPlucked.cpp:19:
/home/pere/src/sonic-pi/pkg-supercollider-sc3-plugins/source/DWGUGens/dwglib/DWG.hpp:1271:53: note:   mismatched types 'std::initializer_list<_Tp>' and 'float'
             loss = 1.0 - std::min(lossF/sqrt(a1),1.0);
                                                     ^
```

This pull request changes the wrong 1.0 value.  The problem is the arguments to std::min() and a fix is to replace the last 1.0 with 1.0f.

**Comment by sonoro1234** [comment]
@snappizz 
Thanks for taking care. I dont know how to continue this. Perhaps is better that you finish that instead of me for not overlapping PRs.
```
loss = 1.0 - std::min(lossF/sqrt(a1),1.0f);
```

**Comment by snappizz** [comment]
according to @petterreinholdtsen's comment, this PR is wrong so I'll start over.

**Comment by snappizz** [comment]
ok, #112 should do it.
